cmsimple/classes/Page.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace XH;
+
+/**
+ * Initiates a page object. Its methods act as a wrapper
+ * of the XH/Pages methods for better OOP access
+ *
+ * @category CMSimple_XH
+ * @package  XH
+ * @author   The CMSimple_XH developers <devs@cmsimple-xh.org>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.en.html GNU GPLv3
+ * @link     http://cmsimple-xh.org/
+ * @since    1.8
+ */
+
+Class Page
+{
+    
+    /**
+     * Instance of XH/Pages.
+     *
+     * @var pages
+     *
+     */
+    private $pages;
+
+    /**
+     * Index selected page.
+     *
+     * @var selected
+     *
+     */
+    private $selected;
+
+    /**
+     * Constructs an instance.
+     *
+     * @param obj pages instance
+     *
+     * @param int $selected Intex of selected page
+     *
+     * @global int Index selected page.
+     */
+    public function __construct($pages,$selected = NULL)
+    {
+        global $s;
+        
+        $this->pages = $pages;
+
+        $this->selected = isset($selected)? $selected: $s;
+    }
+
+    /**
+     * Returns the page object of parent page.
+     *
+     * @return obj Object of the parent page.
+     */
+    public function getParent()
+    {
+        $parent = $this->pages->parent($this->selected);
```

Comment:
This would restrict this method to work for the selected page only. A more generic solution would be to store the page index instead of `XH\Page:$selected` and to use it here and elsewere in this class.

Reply:
Its the page selected in XH\Pages::getSelected(). 
How about to set the page Id by parameter? (XH\Page::getSelected($pageId = null). No parameter means actual Page.

Reply:
Makes sense! I wouldn't call the method `getSelected()`, though. At least traditionally “selected” refers to the *currently* selected (aka. “actual”) page (cf. `$s`, `$su`). What about `::get()`, `::getPage()`, `::getPageAt()` or something like this, instead? And perhaps it might be a good idea to not introduce a default (the “selected” page), since this may not be known exactly when the method is called (cf. `$s` vs. `$pd_s`) (I'm aware that I suggested otherwise recently, but maybe it's too early for such convenience).

Reply:
> ... ), since this may not be known exactly when the method is called (cf. $s vs. $pd_s)

Does this mean that one has to use $pd_s during plugin loading and after use $s? Is $pd_s not valid anymore after plugin loading?